### PR TITLE
Unit Tests for Multiprocessor Snowfakery

### PR DESCRIPTION
**Pull request by prescod**
# Changes

Add tests to increase coverage of Snowfakery to 100%.

Also remove dead code and a few other stylistic improvements:

 * e.g. get_debug_mode() is captured just once as an instance attribute.
 * dead code is removed in some places.
 * a few assertion or warning branches are covered with pragmas.

### CODE REVIEW

**Review by jstvz — APPROVED** [review]
Some quibbles

File: cumulusci/tasks/bulkdata/snowfakery.py [hunk 1]
```diff
@@ -195,8 +196,6 @@ def _run_task(self):
             # tested and documented.
 
             # Retrieve relevant code from f2866e4b
-            self.org_record_counts_thread = None
```
[the remark this answers is not in the document]
Comment (prescod) [reply cumulusci/tasks/bulkdata/snowfakery.py:198]: Clarified in fcb953a9

File: cumulusci/tasks/bulkdata/snowfakery.py [hunk 2]
```diff
@@ -318,12 +317,13 @@ def _report_status(
...
-        ):
-            for k, v in org_record_counts_thread.other_inaccurate_record_counts.items():
-                self.logger.info(f"      COUNT: {k}: {v:,}")
+        # disabled for now
+        # if (
+        #     org_record_counts_thread
+        #     and org_record_counts_thread.other_inaccurate_record_counts
+        # ):
+        #     for k, v in org_record_counts_thread.other_inaccurate_record_counts.items():
+        #         self.logger.info(f"      COUNT: {k}: {v:,}")
```
[the remark this answers is not in the document]
Comment (prescod) [reply cumulusci/tasks/bulkdata/snowfakery.py]: Done in fcb953a9fe8

File: cumulusci/tasks/bulkdata/tests/test_snowfakery.py [hunk 3]
```diff
@@ -133,6 +162,40 @@ def test_run_until_loaded_2_parts(
...
+    #         MyOrgConfig.__getattr__,
+    #     ), mock.patch.object(
+    #         task.org_config,
+    #         "refresh_oauth_token",
+    #         lambda *args, **kwargs: original_refresh_token(
+    #             task.org_config, *args, **kwargs
+    #         ),
+    #     ):
+    #         task()
+
```
[the remark this answers is not in the document]
Comment (prescod) [reply cumulusci/tasks/bulkdata/tests/test_snowfakery.py]: Done in bd425bcbbd33

**Review by jstvz — APPROVED** [review]

**Pull request merged**